### PR TITLE
chore(scanner): Improve indexer config readability

### PR DESCRIPTION
**Pull request by jvdm**
## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

```yaml
http_listen_addr: 127.0.0.1:9443
grpc_listen_addr: 127.0.0.1:8443
indexer:
  enable: true
  database:
    conn_string: "host=/var/run/postgresql"
    password_file: ""
  get_layer_timeout: 1m
  repository_to_cpe_file: "repository-to-cpe.json"
  name_to_repos_file: "container-name-repos-map.json"
  repository_to_cpe_url: "http://localhost:8000/repository-to-cpe.json"
  name_to_repos_url: "http://localhost:8000/container-name-repos-map.json"
matcher:
  enable: false
  database:
    conn_string: "host=/var/run/postgresql"
    password_file: ""
mtls:
  certs_dir: "certs/scanner-v4"
log_level: debug
```

```sh
⮩ python -m http.server                                                  [~/wt/github.com/stackrox/stackrox/health/scanner]
Serving HTTP on 0.0.0.0 port 8000 (http://0.0.0.0:8000/) ...
127.0.0.1 - - [19/Jan/2024 14:46:56] "GET /container-name-repos-map.json HTTP/1.1" 200 -
127.0.0.1 - - [19/Jan/2024 14:46:56] "GET /repository-to-cpe.json HTTP/1.1" 200 -
```

```
{"level":"debug","host":"descending","component":"rhel/internal/common/Updater.Fetch","version":"1.1","url":"http://localhost:8000/repository-to-cpe.json","time":"2024-01-19T14:46:56-03:00","message":"attempting fetch of mapping file"}
{"level":"debug","host":"descending","component":"rhel/internal/common/Updater.Fetch","version":"1.1","url":"http://localhost:8000/repository-to-cpe.json","time":"2024-01-19T14:46:56-03:00","message":"atomic update of local mapping file complete"}
```

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: scanner/indexer/indexer.go [hunk 1]
```diff
@@ -142,7 +141,19 @@ func NewIndexer(ctx context.Context, cfg config.IndexerConfig) (Indexer, error)
...
+		if !ok {
+			return errors.New("internal erorr: casting failed")
+		}
+		f(cfg)
+		return nil
+	}
+}
+
 func newLibindex(ctx context.Context, indexerCfg config.IndexerConfig, root string, store ccindexer.Store, locker *ctxlock.Locker) (*libindex.Libindex, error) {
+
```
Comment (RTann) [inline scanner/indexer/indexer.go:156]: nit: was this on purpose?

File: scanner/indexer/indexer.go [hunk 2]
```diff
@@ -142,7 +141,19 @@ func NewIndexer(ctx context.Context, cfg config.IndexerConfig) (Indexer, error)
 	}, nil
 }
 
+func castToConfig[T any](f func(cfg T)) func(o interface{}) error {
```
Comment (RTann) [inline scanner/indexer/indexer.go:144]: nit: up to you, but `any` is the more "modern" version of `interface{}` and it interchangeable. Up to you if you want to keep this or change to `any`

**Review by RTann — APPROVED** [review]

File: scanner/indexer/indexer.go (hunk 2 above)
Comment (jvdm) [reply scanner/indexer/indexer.go:144]: I think it makes sense to move to `any`.

**Review by RTann — COMMENTED** [review]
want to change all `interface{}` to `any`?

**Pull request merged**